README.md

```diff
@@ -1,10 +1,458 @@
+  Setting the name here will override this.
+
+## Worker environments
+
+A Worker config file may contain configuration for multiple [environments](https://developers.cloudflare.com/workers/wrangler/environments/).
+With the Cloudflare Vite plugin, you select an environment at dev or build time by providing the `CLOUDFLARE_ENV` environment variable.
+Consider the following example `wrangler.toml` file:
+
+```toml
+# wrangler.toml
+
+name = "my-worker"
+compatibility_date = "2024-12-30"
+main = "./src/index.ts"
+
+vars = { MY_VAR = "Top-level var" }
+
+[env.staging]
+vars = { MY_VAR = "Staging var" }
+
+[env.production]
+vars = { MY_VAR = "Production var" }
+```
+
+If you run `CLOUDFLARE_ENV=production vite build` then the output `wrangler.json` file generated by the build will be a flattened configuration for the 'production' environment.
+This combines [top-level only](https://developers.cloudflare.com/workers/wrangler/configuration/#top-level-only-keys), [inheritable](https://developers.cloudflare.com/workers/wrangler/configuration/#inheritable-keys), and [non-inheritable](https://developers.cloudflare.com/workers/wrangler/configuration/#non-inheritable-keys) keys.
+The value of `MY_VAR` will therefore be `'Production var'`.
+The name of the Worker will be `'my-worker-production'`
+This is because the environment name is automatically appended to the top-level Worker name.
+
+> [!NOTE]
+> The default Vite environment name for a Worker is always the top-level name.
+> This enables you to reference the Worker consistently in your Vite config when using multiple environments.
+
+Running `vite dev` or `vite build` without providing `CLOUDFLARE_ENV` will use the default top-level environment.
+The value of `MY_VAR` will therefore be `'Top-level var'`.
+As Worker environments are applied at dev and build time, specifying `CLOUDFLARE_ENV` when running `vite preview` or `wrangler deploy` will have no effect.
+
+### Combining Worker environments and Vite modes
+
+You may wish to combine the concepts of [Worker environments](https://developers.cloudflare.com/workers/wrangler/environments/) and [Vite modes](https://vite.dev/guide/env-and-mode.html#modes).
+With this approach, the mode can be used to select the environment and a single method can be used to determine environment specific configuration and code.
+Consider again the previous example:
+
+```toml
+# wrangler.toml
+
+name = "my-worker"
+compatibility_date = "2024-12-30"
+main = "./src/index.ts"
+
+vars = { MY_VAR = "Top-level var" }
+
+[env.staging]
+vars = { MY_VAR = "Staging var" }
+
+[env.production]
+vars = { MY_VAR = "Production var" }
+```
+
+Next, provide `.env.staging` and `.env.production` files:
+
+```sh
+# .env.staging
+
+CLOUDFLARE_ENV=staging
+```
+
+```sh
+# .env.production
+
+CLOUDFLARE_ENV=production
 ```
+
+Now, if you run `vite build`, the 'production' mode will be used by default and the 'production' environment will be selected.
```

Comment:
I think this could be more explicit, especially for people who don't understand modes and environments.

Perhaps along the lines of:

```suggestion
By default the `vite build`, uses the 'production' mode. Vite will therefore load the `.env.production` file to get environment variables to be used in the build. Since the `.env.production` file contains `CLOUDFLARE_ENV=production` the Cloudflare Vite plugin will select the 'production' Worker environment configuration.
```

Reply:
[e0d0a21](https://github.com/flarelabs-net/vite-plugin-cloudflare/pull/119/commits/e0d0a2124631758d957adc445e01201ed29832a8)